test/BUILD

```diff
@@ -1,49 +1,61 @@
-[sh_test(
-    name = test,
-    # top showed me 50-400mByte of usage, so "enormous" for
-    # long running tests, but the OpenROAD tests are generally
-    # "small" by design.
-    #
-    # https://bazel.build/reference/be/common-definitions#test.size
-    size = "small",
-    timeout = "moderate",
-    srcs = ["bazel_test.sh"],
-    args = [],
-    data = [
-        "//:openroad",
-    ] +
+load("//test:regression.bzl", "regression_test")
+
+exports_files(
+    [
+        "bazel_test.sh",
+        "regression_test.sh",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+filegroup(
+    name = "regression_resources",
     # TODO Refine glob later.
     #
     # This is a very broad glob, but most of the data in size
     # comes from .lib files and we can't know which pdk files
     # to include for a test given the name only.
-    glob([
-        "**/*",
-    ]),
-    env = {
-        "TEST_NAME_BAZEL": test,
-        "OPENROAD_EXE": "$(location //:openroad)",
-    },
-) for test in [
-    # From CMakeLists.txt or_integration_tests(TESTS
-    "error1",
-    "get_core_die_areas",
-    "place_inst",
-    "upf_test",
-    "upf_aes",
-    # Bazel OpenROAD is not compiled with Python support yet
-    #
-    # "timing_api",
-    # "timing_api_2",
-    # "timing_api_3",
-    # "timing_api_4",
-    # "two_designs",
-    #
-    # TODO There are tests in this folder that were not enabled in
-    # CMakeLists.txt, presumably because they are long running.
-    # These should be added with tags = ["manual"], i.e not
-    # run with 'bazel test ...', only when invoked explicitly
-    # and the "size" set to "enormous".
-    #
-    #"aes_nangate45",
-]]
+    srcs = glob(["**/*"]),
+    visibility = ["//visibility:public"],
+)
+
+[
+    regression_test(
+        name = name,
+        tags = kwargs.get("tags", []),
+    )
+    for name, kwargs in (
```

Comment:
this does look somewhat complicated to me.

why not have something like

```python
   tags = ["manual"] if test in ["timing_api", "...and soo on"] else []
   size = "enormous" if test in ["aes_nangate45"] else "normal"
```

That would make it much more immediately readable without having to understand three merged hash-maps generated by list-comprehension.

Reply:
Only add flexibility when needed. Readability is the key. Apply the KISS principle.

Part of using bazel is to have not the incomprehensible mess that cmake files require users to endure; so we're not in a race to match that :)

Reply:
fixed